src/main/java/com/globallogic/amcr/persistence/dao/librarycomponent/BookDao.java

```diff
@@ -0,0 +1,78 @@
+package com.globallogic.amcr.persistence.dao.librarycomponent;
+
+import java.util.List;
+import java.util.UUID;
+
+import com.globallogic.amcr.persistence.dao.Dao;
+import com.globallogic.amcr.persistence.model.librarycomponent.Book;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Repository;
+
+import com.globallogic.amcr.persistence.mapper.librarycomponent.BookMapper;
+
+@Repository
+public class BookDao implements Dao<Book, Book> {
+    public static final Logger Log = LoggerFactory.getLogger(BookDao.class.getName());
+    final BookMapper bookMapper;
+
+    public BookDao(BookMapper bookMapper) {
+        this.bookMapper = bookMapper;
+    }
+
+    public Book save(Book book, UUID id) {
```

Comment:
Assert that these params are not null. this applies to all methods in the service